pkg/controller/nstemplateset/nstemplatetier_test.go

```diff
@@ -252,12 +252,8 @@ func TestGetTierTemplate(t *testing.T) {
 	})
 }
 
-func resetCache() func() {
-	reset := func() {
-		tierTemplatesCache = newTierTemplateCache()
-	}
-	reset()
-	return reset
+func resetCache() {
+	tierTemplatesCache = newTierTemplateCache()
```

Comment:
the original idea was to call reset immediately (in case it wasn't reset before - for example someone could forget to do that at the end of the test) and then also at the end of the test

Reply:
to be honest, the fact that we have `defer resetCache()` in the test did not make it obvious that the cache was also reset at the beginning of the test :(

Reply:
I know, it was confusing... keep it as it is.